assets/js/controllers/buysell/buy.controller.js

```diff
@@ -133,45 +135,25 @@ function BuyCtrl ($scope, $filter, $q, MyWallet, Wallet, MyWalletHelpers, Alerts
     $scope.quote = null;
     $scope.transaction.btc = 0;
     $scope.status.gettingQuote = true;
+    $scope.status.waiting = true;
     if (!$scope.transaction.fiat) { $scope.status = {}; return; }
 
-    let quote;
-    if ($scope.userHasExchangeAcct) {
-      quote = $scope.exchange.getBuyQuote(Math.trunc($scope.transaction.fiat * 100), $scope.transaction.currency.code);
-    } else {
-      quote = exchange.getBuyQuote(Math.trunc($scope.transaction.fiat * 100), $scope.transaction.currency.code);
-    }
-
-    let getPaymentMethods = (quote) => {
-      const getPaymentMethodsError = () => {
-        $scope.standardError({message: 'ERROR_PAYMENT_METHODS_FETCH'});
-      };
-
-      return quote.getPaymentMethods().catch(getPaymentMethodsError);
-    };
-
-    const setQuote = (quote) => {
-      $scope.quote = quote;
-      return quote;
-    };
+    let quoteError = eventualError('ERROR_QUOTE_FETCH');
+    let amount = Math.round($scope.transaction.fiat * 100);
```

Comment:
It might cause this assertion to fail at some point: https://github.com/blockchain/My-Wallet-V3/blob/master/src/coinify/coinify.js#L242

For example: `Math.round(35.05 * 100) === 35.05 * 100 // false`